numpy/core/_type_aliases.py

```diff
@@ -1,224 +1,263 @@
+    def _process_integer_type(
+        name: str, 
+        info: typeinforanged, 
+        priority_dict: Dict[str, int], 
+        seen_bits_dict: Dict[int, str],
+        long_name: str, 
+        short_name: str
+    ) -> STATUS:
+        bits: int = info.bits 
+        if name in priority_dict and (
+            bits not in seen_bits_dict or 
+            priority_dict[seen_bits_dict[bits]] > priority_dict[name]
+        ):
+            word_bits_dict[f"{long_name}{bits}"] = info.type
+            symbol_bytes_dict[f"{short_name}{bits//8}"] = info.type
+            seen_bits_dict[bits] = name
+            return PROCESSED
+        else:
+            return REJECTED
+
+    # We will track which sizes we have already seen for ints and uints
+    _seen_int_bits = {}
+    _seen_uint_bits = {}
+
+    # MAIN LOOP: Traversing all contrete types that come from 
+    # the compiled multiarray module
     for name, info in _concrete_typeinfo.items():
-        # these are handled by _add_integer_aliases
-        if name in _int_ctypes or name in _uint_ctypes:
+        
+        bits: int = info.bits
+        symbol: str = info.char
+        
+        # Adding type to non-bit dictionaries
+        word_dict[name] = info.type
+        symbol_dict[symbol] = info.type
+        numbers_dict[info.num] = info.type
+
+        # Adding type to bit-aware dictionaries
+        # 1. Checking if the type is a signed integer
+        result = _process_integer_type(
+            name, info, _int_ctypes_prio, _seen_int_bits, "int", "i"
+        )
+        if result is PROCESSED:
             continue
-
-        # insert bit-width version for this class (if relevant)
-        base, bit, char = bitname(info.type)
-
-        myname = "%s%d" % (base, bit)
-
-        # ensure that (c)longdouble does not overwrite the aliases assigned to
-        # (c)double
-        if name in ('longdouble', 'clongdouble') and myname in allTypes:
+        # 2. If not, then we check if it's an unsigned integer
+        result = _process_integer_type(
+            name, info, _uint_ctypes_prio, _seen_uint_bits, "uint", "u"
+        )
+        if result is PROCESSED:
             continue
+        if name in _complex_ctypes + _floating_ctypes + _other_types:
+            # 3. Otherwise it can be float/complex or other type
+            dt = dtype(info.type)
+            word_bits_name = f"{_kind_name(dt)}{bits}"
+            symbol_bytes_name = f"{dt.kind}{bits//8}"
+
+            # ensure that (c)longdouble does not overwrite the aliases 
+            # assigned to (c)double, when both have same number of bits
+            if (
+                name in ('longdouble', 'clongdouble') and
+                word_bits_name in word_bits_dict
+            ):
+                continue
+
+            word_bits_dict[word_bits_name] = info.type
+            symbol_bytes_dict[symbol_bytes_name] = info.type
+
+
+_build_dicts()
+
+
+# Rename types to Python conventions and introduce aliases
```

Comment:
This comment can also be set as a docstring for `_renaming_and,_aliases`

Reply:
Done!